src/main.rs

```diff
@@ -0,0 +1,106 @@
+use std::io::{self, Write};
+use regex::{self, Regex};
+
+enum MetaCommand{
+    Exit
+}
+
+impl MetaCommand{
+    fn is_meta_command(input_str: &str) -> bool {
+        input_str.starts_with('.')
+    }
+
+    fn categorize(input_str: &str) -> Option<Self>{
+        match input_str {
+            ".exit" => Some(MetaCommand::Exit),
+            _       => None
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+enum Statement {
+    Insert,
+    Select
+}
+
+#[derive(Debug)]
+enum ParseError{
+    RegexCreation(regex::Error),
+    NoRegexMatch
+}
+
+impl From<regex::Error> for ParseError {
+    fn from(err: regex::Error) -> ParseError {
+        ParseError::RegexCreation(err)
+    }
+}
+
+#[derive(Debug)]
+struct Schema{
+    id: u8,
+    username: [char; 32],
+    email: [char; 32]
+}
+
+fn to_fixlen_array(input: std::str::Chars) -> [char; 32] {
+    // TODO: Better way? Also handle overflow case
+    let mut result = ['\0'; 32];
+    for (i, c) in input.enumerate() {
+        result[i] = c;
+    }
+    result
+}
```

Comment:
Here's something I came up with. The problem is... I don't think you can handle overflow in any way with this approach.

```suggestion
fn to_fixlen_array(input: std::str::Chars) -> [char; 32] {
    let vec: Vec<char> = input.chain(repeat('\0')).take(32).collect();
    vec.try_into().unwrap()
}
```

Generally speaking I don't think taking `std::str::Chars` as an argument is a good idea. You're better off using `&str`. Just change the signature here and pass `&captures[whatever]` in the calling code. That way you could also use `.len()` on the string slice argument.